src/main/g8/test/connectors/MongoCacheConnectorSpec.scala

```diff
@@ -0,0 +1,198 @@
+        val mockSessionRepository = mock[SessionRepository]
+
+        when(mockSessionRepository.apply()) thenReturn mockReactiveMongoRepository
+        when(mockReactiveMongoRepository.get(any())) thenReturn Future(None)
+
+        val mongoCacheConnector = new MongoCacheConnector(mockSessionRepository)
+
+        forAll(nonEmptyString) {
+          cacheId =>
+
+            val result = mongoCacheConnector.fetch(cacheId)
+
+            whenReady(result) {
+              optionalCacheMap =>
+
+                optionalCacheMap must be(empty)
+            }
+        }
+      }
+    }
+
+    "a record exists for this key" must {
+
+      "return the record" in {
+
+        val mockReactiveMongoRepository = mock[ReactiveMongoRepository]
+        val mockSessionRepository = mock[SessionRepository]
+
+        when(mockSessionRepository.apply()) thenReturn mockReactiveMongoRepository
+
+        val mongoCacheConnector = new MongoCacheConnector(mockSessionRepository)
+
+        forAll(arbitrary[CacheMap]) {
+          cacheMap =>
+
+            when(mockReactiveMongoRepository.get(eqTo(cacheMap.id))) thenReturn Future(Some(cacheMap))
+
+            val result = mongoCacheConnector.fetch(cacheMap.id)
+
+            whenReady(result) {
+              optionalCacheMap =>
+
+                optionalCacheMap.value mustEqual cacheMap
+            }
+        }
+      }
+    }
+  }
+
+  ".getEntry" when {
+
+    "there isn't a record for this key in Mongo" must {
+
+      "return None" in {
+
+        val mockReactiveMongoRepository = mock[ReactiveMongoRepository]
+        val mockSessionRepository = mock[SessionRepository]
+
+        when(mockSessionRepository.apply()) thenReturn mockReactiveMongoRepository
+        when(mockReactiveMongoRepository.get(any())) thenReturn Future(None)
+
+        val mongoCacheConnector = new MongoCacheConnector(mockSessionRepository)
+
+        forAll(nonEmptyString, nonEmptyString) {
+          (cacheId, key) =>
+
+            val result = mongoCacheConnector.getEntry[String](cacheId, key)
+
+            whenReady(result) {
+              optionalValue =>
+
+                optionalValue must be(empty)
+            }
+        }
+      }
+    }
+
+    "a record exists in Mongo but this key is not present" must {
+
+      "return None" in {
+
+        val mockReactiveMongoRepository = mock[ReactiveMongoRepository]
+        val mockSessionRepository = mock[SessionRepository]
+
+        when(mockSessionRepository.apply()) thenReturn mockReactiveMongoRepository
+
+        val mongoCacheConnector = new MongoCacheConnector(mockSessionRepository)
+
+        val gen = for {
+          key      <- nonEmptyString
+          cacheMap <- arbitrary[CacheMap]
+        } yield (key, cacheMap copy (data = cacheMap.data - key))
+
+        forAll(gen) {
+          case (key, cacheMap) =>
+
+            when(mockReactiveMongoRepository.get(eqTo(cacheMap.id))) thenReturn Future(Some(cacheMap))
```

Comment:
Please use `Future.successful( )` instead of `Future`

Reply:
Thanks, will do